src/Entity/Trick.php

```diff
@@ -0,0 +1,167 @@
+     * @ORM\Id()
+     * @ORM\GeneratedValue
+     * @ORM\Column(type="integer")
+     */
+    private $id;
+
+    /**
+     * @ORM\Column(type="string", length=255)
+     */
+    private $name;
+
+    /**
+     * @ORM\Column(type="text")
+     */
+    private $chapo;
+
+    /**
+     * @ORM\Column(type="text")
+     */
+    private $description;
+
+    /**
+     * @ORM\Column(type="string", length=255, nullable=true)
+     */
+    private $imageLink;
+
+    /**
+     * @ORM\Column(type="string", length=255, nullable=true)
+     */
+    private $videoLink;
+
+    /**
+     * @ORM\ManyToOne(targetEntity="App\Entity\User", inversedBy="tricks")
+     * @ORM\JoinColumn(name="user_id", referencedColumnName="id", nullable=false)
+     */
+    private $user;
+
+    /**
+     * @ORM\ManyToOne(targetEntity="App\Entity\Category", inversedBy="tricks")
+     * @ORM\JoinColumn(name="categorie_ìd", referencedColumnName="id", nullable=false)
+     */
+    private $category;
+
+    /**
+     * @ORM\OneToMany(targetEntity="App\Entity\Comment", mappedBy="trick")
+     */
+    private $comments;
+
+    public function getId(): ?int
+    {
+        return $this->id;
+    }
+
+    public function setId(string $id): self
+    {
+        $this->id = $id;
+
+        return $this;
+    }
+
+    public function getName(): ?string
+    {
+        return $this->name;
+    }
+
+    public function setName(string $name): self
+    {
+        $this->name = $name;
+
+        return $this;
+    }
+
+    public function getChapo(): ?string
+    {
+        return $this->chapo;
+    }
+
+    public function setChapo(string $chapo): self
+    {
+        $this->chapo = $chapo;
+
+        return $this;
+    }
+
+    public function getDescription(): ?string
+    {
+        return $this->description;
+    }
+
+    public function setDescription(string $description): self
+    {
+        $this->description = $description;
+
+        return $this;
+    }
+
+    public function getImageLink(): ?string
+    {
+        return $this->imageLink;
+    }
+
+    public function setImageLink(?string $imageLink): self
+    {
+        $this->imageLink = $imageLink;
+
+        return $this;
+    }
+
+    public function getVideoLink(): ?string
+    {
+        return $this->videoLink;
+    }
+
+    public function setVideoLink(?string $videoLink): self
+    {
+        $this->videoLink = $videoLink;
+
+        return $this;
+    }
+
+    public function getUser()
+    {
+        return $this->user;
+    }
+
+    /**
+     * @param mixed $user
+     */
+    public function setUser ( $user )
+    {
+        $this->user = $user;
+
+        return $user;
+    }
+
+    public function getCategory()
+    {
+        return $this->category;
+    }
+
+    /**
+     * @param mixed $category
+     */
+    public function setCategory ($category)
+    {
+        $this->category = $category;
+
+        return $category;
+    }
+
+    public function getComments ()
+    {
+        return $this->comments;
+    }
```

Comment:
Sur les oneToMany, en général, en plus des getter et setter, on ajoute les adder et remover. Je t'invite à aller voir de quoi je parle et les rajouter dans sur les attributs oneToMany de toutes tes entités ;)

Reply:
Fait. 
_src = https://symfony.com/doc/current/doctrine/associations.html#associations-inverse-side_

Reply:
nikel

Reply:
euh en fait, non pas nikel.

$comments n'est pas un tableau mais un ArrayCollection. De ce fait, pour ajouter un element, c'est $this->comments->add($comment);

Reply:
Corrigé pour tous les OneToMany.

Reply:
Pour les adders, j'ai consulté la documentation et j'ai fait pour tous les OneToMany : 
```
/**
 * @ORM\OneToMany(targetEntity="App\Entity\Comment", mappedBy="trick")
 */
private $comments;

public function __construct ()
{
    $this->comments = new ArrayCollection();
}
public function getComments ()
{
    return $this->comments;
}

public function addComment(Comment $comment):self
{
    if ($this->comments->contains($comment)) {
        $this->comments->add($comment);
        $comment->setTrick($this);
    }

    return $this;
}

public function removeComment (Comment $comment):self
{
    if($this->comments->contains($comment)) {
        $this->comments->removeElement($comment);
        if($comment->getTrick() === $this){
            $comment->setTrick(null);
        }
    }

    return $this;
}
```

Ce n'était pas ce que tu entendais par adders ?